lib/core/test/bench/db/Main.hs

```diff
@@ -503,17 +503,28 @@ benchPutSeqState :: Int -> Int -> DBLayerBench -> IO ()
 benchPutSeqState numCheckpoints numAddrs DBLayer{..} =
     unsafeRunExceptT $ mapM_ (mapExceptT atomically . putCheckpoint testPk)
         [ snd $ initWallet block0 genesisParameters $
-            SeqState (mkPool numAddrs i) (mkPool numAddrs i) emptyPendingIxs
+            SeqState
+                (mkPool numAddrs i)
+                (mkPool numAddrs i)
+                emptyPendingIxs
+                rewardAccount
         | i <- [1..numCheckpoints]
         ]
 
 mkPool
-    :: forall t c. (PaymentAddress t ShelleyKey, Typeable c)
-    => Int -> Int -> AddressPool t c ShelleyKey
+    :: forall c. (Typeable c)
+    => Int -> Int -> AddressPool c ShelleyKey
 mkPool numAddrs i = mkAddressPool ourAccount defaultAddressPoolGap addrs
   where
     addrs = [ mkAddress i j | j <- [1..numAddrs] ]
 
+rewardAccount
+    :: ShelleyKey 'AddressK XPub
+rewardAccount =
+    ShelleyKey $ unsafeFromRight $ xpub $ BS.replicate 64 0
+  where
+    unsafeFromRight = either (error "unsafeFromRight: invalid xpub?") id
```

Comment:
do we have test for that?

Reply:
It's not really possible in practice to get an invalid `XPub` because they all come from our derivation functions and from the root -> `generateKeyFromSeed`

Here, I am using the unsafe constructor `xpub` although I could actually just use `unsafeGenerateKeyFromSeed` :thinking: ...

Reply:
Fixed in https://github.com/input-output-hk/cardano-wallet/pull/1073/commits/a3c8973fa7cc7a1006adda8926b72612d0f94be7